### PR TITLE
Stop building gcc from source

### PR DESCRIPTION
**Pull request by carlzogh**
*Description of changes:*
This PR:
- Changes the RIC's build process to use gcc from `yum` as opposed to building it from source (as it's now a modern supported version after updating to CentOS 7 done [here](https://github.com/aws/aws-lambda-java-libs/pull/193))
- Removes unnecessary `user.dir` overrides in `CustomerClassLoaderTest`
- Adds glibc OS test in test-smoke (now also testing `amazoncorretto` version `11` - previously we were only testing Alpine with musl)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by stefanbotez — APPROVED** [review]
LGTM!

File: aws-lambda-java-runtime-interface-client/src/test/java/com/amazonaws/services/lambda/runtime/api/client/CustomerClassLoaderTest.java [hunk 1]
```diff
@@ -20,18 +18,6 @@
 import static org.junit.jupiter.api.condition.OS.MAC;
 
 public class CustomerClassLoaderTest {
-    private static final String USER_WORKING_DIR = "user/path";
```
Comment (edhzsz) [inline aws-lambda-java-runtime-interface-client/src/test/java/com/amazonaws/services/lambda/runtime/api/client/CustomerClassLoaderTest.java:23]: Feels like this changes should had been done in a different PR.

**Review by edhzsz — APPROVED** [review]
